newIDE/app/src/MainFrame/index.js

```diff
@@ -142,6 +142,7 @@ type State = {|
 type Props = {
   integratedEditor?: boolean,
   introDialog?: React.Element<*>,
+  mainMenu?: Class<React$Component<*, *>>,
```

Comment:
I would still favor a `renderMainMenu` prop.
Feel free to put the props in a separate type (`renderMainMenu: (MainMenuProps) => React.Node`).

It's ok because it's MainFrame that must dictate what the main menu must support :) 
You can export the type (`export type MainMenuProps = {| ... |};`) so that ElectronMainMenu can import it and use it for its props. So that you're always sure things are 100% in sync! :) 

In the future, if we add a main menu in the web-app, it will be very easy to just use the same type. This is clean and explicit :)

Reply:
(the * in Flow is actually a hidden "any" I think, which does not give us any type checking on the props here)